pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -102,213 +178,381 @@ func (mi *StorageInitializerInjector) InjectModelcar(pod *corev1.Pod) error {
+			err := params.CredentialBuilder.CreateSecretVolumeAndEnv(
+				ctx,
+				params.Namespace,
+				params.IsvcAnnotations,
+				params.PodSpec.ServiceAccountName,
+				initContainer,
+				&params.PodSpec.Volumes,
+			)
+			if err != nil {
+				return err
 			}
 		}
 
-		initContainer.Env = append(initContainer.Env, corev1.EnvVar{
-			Name:  constants.CaBundleConfigMapNameEnvVarKey,
-			Value: caBundleConfigMapName,
-		})
+		// Inject CA bundle configMap if caBundleConfigMapName or constants.DefaultGlobalCaBundleConfigMapName annotation is set
+		caBundleConfigMapName := params.Config.CaBundleConfigMapName
+		if ok := needCaBundleMount(caBundleConfigMapName, initContainer); ok {
+			if params.Namespace != constants.KServeNamespace {
+				caBundleConfigMapName = constants.DefaultGlobalCaBundleConfigMapName
+			}
 
-		initContainer.Env = append(initContainer.Env, corev1.EnvVar{
-			Name:  constants.CaBundleVolumeMountPathEnvVarKey,
-			Value: caBundleVolumeMountPath,
-		})
+			caBundleVolumeMountPath := params.Config.CaBundleVolumeMountPath
+			if caBundleVolumeMountPath == "" {
+				caBundleVolumeMountPath = constants.DefaultCaBundleVolumeMountPath
+			}
+
+			for _, envVar := range initContainer.Env {
+				if envVar.Name == s3.AWSCABundleConfigMap {
+					caBundleConfigMapName = envVar.Value
+				}
+				if envVar.Name == s3.AWSCABundle {
+					caBundleVolumeMountPath = filepath.Dir(envVar.Value)
+				}
+			}
 
-		caBundleVolume := corev1.Volume{
-			Name: CaBundleVolumeName,
-			VolumeSource: corev1.VolumeSource{
-				ConfigMap: &corev1.ConfigMapVolumeSource{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: caBundleConfigMapName,
+			initContainer.Env = append(initContainer.Env, corev1.EnvVar{
+				Name:  constants.CaBundleConfigMapNameEnvVarKey,
+				Value: caBundleConfigMapName,
+			})
+
+			initContainer.Env = append(initContainer.Env, corev1.EnvVar{
+				Name:  constants.CaBundleVolumeMountPathEnvVarKey,
+				Value: caBundleVolumeMountPath,
+			})
+
+			caBundleVolume := corev1.Volume{
+				Name: CaBundleVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: caBundleConfigMapName,
+						},
 					},
 				},
-			},
+			}
+
+			caBundleVolumeMount := corev1.VolumeMount{
+				Name:      CaBundleVolumeName,
+				MountPath: caBundleVolumeMountPath,
+				ReadOnly:  true,
+			}
+
+			params.PodSpec.Volumes = append(params.PodSpec.Volumes, caBundleVolume)
+			initContainer.VolumeMounts = append(initContainer.VolumeMounts, caBundleVolumeMount)
 		}
 
-		caBundleVolumeMount := corev1.VolumeMount{
-			Name:      CaBundleVolumeName,
-			MountPath: caBundleVolumeMountPath,
-			ReadOnly:  true,
+		// Update initContainer (container spec) from a storage container CR if there is a match,
+		// otherwise initContainer is not updated.
+		// Priority: CR > configMap
```

Comment:
```suggestion
```
Can remove this comment since matching storage container CR is moved to the controller